boot/freeldr/uefi/uefisetup.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * PROJECT:     Freeldr UEFI Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     UEFI Mach Setup
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+#include <uefildr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(WARNING);
+
+/* GLOBALS ********************************************************************/
+
+EFI_GUID EfiGraphicsOutputProtocol = EFI_GRAPHICS_OUTPUT_PROTOCOL_GUID;
+extern EFI_SYSTEM_TABLE *GlobalSystemTable;
+extern EFI_HANDLE GlobalImageHandle;
+EFI_GRAPHICS_OUTPUT_PROTOCOL* gop;
+BOOLEAN AcpiPresent = FALSE;
+
+/* FUNCTIONS ******************************************************************/
+
+EFI_STATUS
+UefiHandleService(EFI_STATUS Status)
+{
+    if (Status != EFI_SUCCESS)
+    {
+        TRACE("UEFI Services call has failed with status: %X\n", Status);
+    }
+
+    return Status;
+}
+
+VOID
+MachInit(const char *CmdLine)
+{
+    EFI_STATUS Status = 0;
+    Status = GlobalSystemTable->BootServices->AllocatePool(EfiLoaderData, sizeof(MachVtbl), (void**)&MachVtbl);
+    if (UefiHandleService(Status) != EFI_SUCCESS)
+        ERR("UefiMachInit: Could not allocated memory for MachVtbl");
+
+    /* Setup GOP */
+    Status = GlobalSystemTable->BootServices->LocateProtocol(&EfiGraphicsOutputProtocol, 0, (void**)&gop);
+    if (Status != EFI_SUCCESS)
+        ERR("UefiMachInit: Cannot setup GOP");
```

Comment:
If you want to follow what was done also for XBOX and PC98 (see `XboxVideoInit` and `Pc98VideoInit`), this gop setup could be moved into the UefiInitializeVideo function. (If so you could make that latter one "VOID"; it wouldn't require the gop pointer as parameter.)